delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3FileOperations.scala

```diff
@@ -63,6 +74,34 @@ object S3FileOperations {
 
     override def save(storage: S3Storage, filename: String, entity: BodyPartEntity): IO[FileStorageMetadata] =
       saveFile.apply(storage, filename, entity)
+
+    override def register(bucket: String, path: Uri.Path): IO[S3FileMetadata] = {
+      for {
+        _           <- log.info(s"Fetching attributes for S3 file. Bucket $bucket at path $path")
+        resp        <- client.headObject(bucket, path.toString())
+        contentType <- parseContentType(resp.contentType())
+      } yield mkS3Metadata(bucket, path, resp, contentType)
+    }
+      .onError { e =>
+        log.error(e)(s"Failed fetching required attributes for S3 file registration. Bucket $bucket and path $path")
+      }
+
+    private def parseContentType(raw: String): IO[ContentType] =
+      ContentType.parse(raw).map(_.pure[IO]).getOrElse(IO.raiseError(InvalidContentType(raw)))
+
+    private def mkS3Metadata(bucket: String, path: Uri.Path, resp: HeadObjectResponse, ct: ContentType) = {
+      S3FileMetadata(
+        ct,
+        FileStorageMetadata(
+          UUID.randomUUID(), // TODO unused field for this use case, but mandatory everywhere?
+          resp.contentLength(),
+          Digest.ComputedDigest(DigestAlgorithm.MD5, resp.eTag().filterNot(_ == '"')),
```

Comment:
It may not be MD5, it is better to check for the checksum part in:
https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObjectAttributes.html